subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidPluginsSmokeTest.groovy

```diff
@@ -30,88 +32,100 @@ import spock.lang.Unroll
  * https://androidstudio.googleblog.com/
  *
  */
+@Requires(TestPrecondition.JDK11_OR_EARLIER)
 class AndroidPluginsSmokeTest extends AbstractSmokeTest {
 
-
     public static final String JAVA_COMPILE_DEPRECATION_MESSAGE = "Extending the JavaCompile task has been deprecated. This is scheduled to be removed in Gradle 7.0. Configure the task instead."
 
     def setup() {
         AndroidHome.assertIsSet()
     }
 
     @Unroll
-    @ToBeFixedForInstantExecution
-    def "android application plugin #pluginVersion"(String pluginVersion) {
+    @UnsupportedWithInstantExecution(iterationMatchers = AGP_3_ITERATION_MATCHER)
+    def "android library and application APK assembly (agp=#agpVersion, ide=#ide)"(
+        String agpVersion, boolean ide
```

Comment:
❓ Are the parameter declarations necessary here or just a matter of style?

Reply:
Not necessary, but it makes their type explicit and helped writing the body of the test without having IDE warnings.

Reply:
Got it.